### PR TITLE
Picture from camera and network status monitoring

### PR DESCRIPTION
**Pull request by robertsmarky**

### CODE REVIEW

File: app/src/main/java/de/tudarmstadt/smartcitystudyapp/ui/incidents/SubmitViewModel.kt [hunk 1]
```diff
@@ -34,23 +35,27 @@ class SubmitViewModel @ViewModelInject constructor(
...
-                message = view.findViewById<EditText>(R.id.report_text).text.toString(),
-                picture = view.findViewById<SwitchCompat>(R.id.switch_send_photo).isChecked,
-                location = view.findViewById<SwitchCompat>(R.id.switch_send_location).isChecked,
-                source = source
-            )
-            reportService.sendReport(report)
-            Toast.makeText(context, R.string.report_sent_toast, Toast.LENGTH_SHORT).show()
-            view.findNavController().navigate(R.id.action_global_home)
-            println("Sent report $report")
+        if(MainActivity.network_status == true) {
```
Comment (robertsmarky) [inline app/src/main/java/de/tudarmstadt/smartcitystudyapp/ui/incidents/SubmitViewModel.kt:40]: Shoudl be just `if(MainActivity.network_status) {` even though it is less readable. For that reason `network_status` should be renamed to `networkAvailable` to fit the actual content and type and to conform with Java's/Kotlins variable name convention.

File: app/src/main/res/layout/fragment_submitincidents.xml [hunk 2]
```diff
@@ -135,20 +135,39 @@
...
+        android:orientation="horizontal">
+
+        <Button
+            android:id="@+id/incidents_button_gallery"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:layout_weight="1"
+            android:background="#00000000"
+            android:text="Gallery"
```
Comment (robertsmarky) [inline app/src/main/res/layout/fragment_submitincidents.xml:150]: Put as a string resource.

File: app/src/main/res/layout/fragment_submitincidents.xml [hunk 3]
```diff
@@ -135,20 +135,39 @@
...
+            android:visibility="invisible" />
+
+        <Button
+            android:id="@+id/incidents_button_camera"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:layout_weight="1"
+            android:background="#00000000"
+            android:text="Kamera"
```
Comment (robertsmarky) [inline app/src/main/res/layout/fragment_submitincidents.xml:161]: Put as a string resource.

File: app/src/main/java/de/tudarmstadt/smartcitystudyapp/MainActivity.kt [hunk 4]
```diff
@@ -59,6 +66,47 @@ class MainActivity : AppCompatActivity() {
...
         drawerLayout.openDrawer(GravityCompat.START)
+
+        // listen for wifi/mobile connectivity changes
+        networkMonitor.result = { isAvailable, type ->
+            runOnUiThread {
+                when (isAvailable) {
+                    true -> {
+                        when (type) {
+                            ConnectionType.Wifi, ConnectionType.Cellular -> {
+                                network_status = true;
```
Comment (robertsmarky) [inline app/src/main/java/de/tudarmstadt/smartcitystudyapp/MainActivity.kt:77]: No need for semicolons in Kotlin.

File: app/src/main/java/de/tudarmstadt/smartcitystudyapp/MainActivity.kt [hunk 5]
```diff
@@ -24,6 +27,10 @@ class MainActivity : AppCompatActivity() {
     @Inject
     lateinit var userService: UserService
     private lateinit var appBarConfiguration: AppBarConfiguration
+    private val networkMonitor = NetworkMonitor(this)
+    companion object {
+        var network_status = true
```
Comment (robertsmarky) [inline app/src/main/java/de/tudarmstadt/smartcitystudyapp/MainActivity.kt:32]: Should be `networkStatus` or even better `networkAvailable`

**Review by robertsmarky — COMMENTED** [review]
The intended feature works: The Camera intent is called correctly, submit button reacts to network changes, and an appropriate toast is shown when attempting to submit a report without network. 
However the regular success toasts are not shown anymore. This needs to be fixed for the merge.
Various other minor comments can also be addressed since they are all quite minor.

File: app/src/main/java/de/tudarmstadt/smartcitystudyapp/ui/incidents/SubmitViewModel.kt [hunk 6]
```diff
@@ -34,23 +35,27 @@ class SubmitViewModel @ViewModelInject constructor(
...
+                Toast.makeText(context, R.string.report_prepare_toast, Toast.LENGTH_SHORT).show()
+
+                val report = Report(
+                    userId = userService.getUserId() ?: "???",
+                    message = view.findViewById<EditText>(R.id.report_text).text.toString(),
+                    picture = view.findViewById<SwitchCompat>(R.id.switch_send_photo).isChecked,
+                    location = view.findViewById<SwitchCompat>(R.id.switch_send_location).isChecked,
+                    source = source
+                )
+                reportService.sendReport(report)
```
[the remark this answers is not in the document]
Comment (th000r) [reply app/src/main/java/de/tudarmstadt/smartcitystudyapp/ui/incidents/SubmitViewModel.kt:53]: I've added another toast which will be displayed if the report was not submitted and logged the result for debugging

File: app/src/main/java/de/tudarmstadt/smartcitystudyapp/ui/incidents/SubmitViewModel.kt [hunk 7]
```diff
@@ -34,23 +35,27 @@ class SubmitViewModel @ViewModelInject constructor(
...
-            )
-            reportService.sendReport(report)
-            Toast.makeText(context, R.string.report_sent_toast, Toast.LENGTH_SHORT).show()
-            view.findNavController().navigate(R.id.action_global_home)
-            println("Sent report $report")
+        if(MainActivity.network_status == true) {
+            println("Sending report") //TODO: Remove debug reports
+            viewModelScope.launch(Dispatchers.IO) {
+                Looper.myLooper() ?: Looper.prepare()
+                Toast.makeText(context, R.string.report_prepare_toast, Toast.LENGTH_SHORT).show()
```
[the remark this answers is not in the document]
Comment (th000r) [reply app/src/main/java/de/tudarmstadt/smartcitystudyapp/ui/incidents/SubmitViewModel.kt:44]: it should work now. the toasts are now executed in the main thread

**Comment by th000r** [comment]
I have updated the code and merged it with the main branch. 
I have tested it on api 29, 21 and 17.  (Sending the report on 17 crashes the app, it think it has to to do with the version of OKHttpClient)

**Pull request merged**